charts/oidc-proxy/templates/configmap.yaml

```diff
@@ -21,11 +21,27 @@ data:
 
     Header unset X-Frame-Options
     Header always set X-Frame-Options SAMEORIGIN
-    Header always set Access-Control-Allow-Origin *
     Header always set Access-Control-Allow-Headers DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization,Accept,Referer,X-App-Id,Origin
     Header always set Access-Control-Allow-Methods GET,POST,DELETE,PUT,PATCH,OPTIONS,HEAD
     Header always set X-Content-Type-Options nosniff
 
+    # If CORS_REGEX environment variable is set, match the domains in the regex
+    # to set the Access-Control-Allow-Origin. Domains not in the regex will be
+    # denied.
+    <If "-n osenv('CORS_REGEX')">
+        SetEnvIf Origin "${CORS_REGEX}" ORIGIN_SUB_DOMAIN=$0
```

Comment:
For what it's worth, I was able to get something working using the Referer header, which seems to more frequently set:
```
SetEnvIfNoCase Referer "^(http(s)?:\\/\\/[^/]+).*$" refhost=$1
SetEnvIf refhost "${CORS_REGEX}" ORIGIN_SUB_DOMAIN=$0
Header set Access-Control-Allow-Origin %{ORIGIN_SUB_DOMAIN}e env=ORIGIN_SUB_DOMAIN
```
This uses a regex to pull the origin part out of the URL

Reply:
I am not sure Referer is more reliable, a user can disable setting Referer on browsers, and there are many situations where it can be empty: https://stackoverflow.com/a/6880668